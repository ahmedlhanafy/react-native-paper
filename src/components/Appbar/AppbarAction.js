/* @flow */

import * as React from 'react';
import IconButton from '../IconButton';
import type { IconSource } from '../Icon';
import type { Theme } from '../../types';
import { withTheme } from '../../core/theming';

type Props = React.ElementConfig<typeof IconButton> & {|
  /**
   *  Custom color for action icon.
   */
  color?: string,
  /**
   * Name of the icon to show.
   */
  icon: IconSource,
  /**
   * Optional icon size.
   */
  size?: number,
  /**
   * Whether the button is disabled. A disabled button is greyed out and `onPress` is not called on touch.
   */
  disabled?: boolean,
  /**
   * Accessibility label for the button. This is read by the screen reader when the user taps the button.
   */
  accessibilityLabel?: string,
  /**
   * Function to execute on press.
   */
  onPress?: () => mixed,
  style?: any,
  /**
   * @optional
   */
  theme: Theme,
|};

/**
 * A component used to display an action item in the appbar.
 */
class AppbarAction extends React.Component<Props> {
  static displayName = 'Appbar.Action';

  static defaultProps = {
    size: 24,
  };

  render() {
    const {
      color: iconColor,
      icon,
      disabled,
      onPress,
      accessibilityLabel,
      theme: { colors },
      ...rest
    } = this.props;

    return (
      <IconButton
        onPress={onPress}
        color={iconColor || colors.typography.secondary}
        icon={icon}
        disabled={disabled}
        accessibilityLabel={accessibilityLabel}
        animated
        {...rest}
      />
    );
  }
}

export default withTheme(AppbarAction);
