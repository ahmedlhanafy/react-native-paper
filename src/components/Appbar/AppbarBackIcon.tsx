import * as React from 'react';
import { Platform, View, Image, StyleSheet } from 'react-native';

import { useLocale } from '../../core/Localization';
import MaterialCommunityIcon from '../MaterialCommunityIcon';

const AppbarBackIcon = ({ size, color }: { size: number; color: string }) => {
  const { direction } = useLocale();
  const iosIconSize = size - 3;

  return Platform.OS === 'ios' ? (
    <View
      style={[
        styles.wrapper,
        {
          width: size,
          height: size,
          transform: [{ scaleX: direction === 'rtl' ? -1 : 1 }],
        },
      ]}
    >
      <Image
        source={require('../../assets/back-chevron.png')}
        style={[
          styles.icon,
          { tintColor: color, width: iosIconSize, height: iosIconSize },
        ]}
        accessibilityIgnoresInvertColors
      />
    </View>
  ) : (
    <MaterialCommunityIcon
      name="arrow-left"
      color={color}
      size={size}
      direction={direction}
    />
  );
};

const styles = StyleSheet.create({
  wrapper: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {
    resizeMode: 'contain',
  },
});

export default AppbarBackIcon;

// @component-docs ignore-next-line
export { AppbarBackIcon };
