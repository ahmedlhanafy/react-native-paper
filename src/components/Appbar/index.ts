import { Appbar as AppbarComponent } from './Appbar';
import AppbarContent from './AppbarContent';
import AppbarAction from './AppbarAction';
import AppbarBackAction from './AppbarBackAction';
import AppbarHeader from './AppbarHeader';

const Appbar = Object.assign(AppbarComponent, {
  // @component ./AppbarContent.tsx
  Content: AppbarContent,
  // @component ./AppbarAction.tsx
  Action: AppbarAction,
  // @component ./AppbarBackAction.tsx
  BackAction: AppbarBackAction,
  // @component ./AppbarHeader.tsx
  Header: AppbarHeader,
});

export default Appbar;
