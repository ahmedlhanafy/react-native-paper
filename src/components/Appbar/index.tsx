import { Appbar as _Appbar } from './Appbar';

import AppbarContent from './AppbarContent';
import AppbarAction from './AppbarAction';
import AppbarBackAction from './AppbarBackAction';
import AppbarHeader from './AppbarHeader';

import { withTheme } from '../../core/theming';

export class Appbar extends _Appbar {
  // @component ./AppbarContent.tsx
  static Content = AppbarContent;
  // @component ./AppbarAction.tsx
  static Action = AppbarAction;
  // @component ./AppbarBackAction.tsx
  static BackAction = AppbarBackAction;
  // @component ./AppbarHeader
  static Header = AppbarHeader;
}

export default withTheme(Appbar);
