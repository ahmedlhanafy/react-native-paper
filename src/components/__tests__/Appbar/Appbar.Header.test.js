import React from 'react';
import renderer from 'react-test-renderer';
import Appbar from '../../Appbar/Appbar';

describe('AppbarHeader', () => {
  it('Uses a SafeAreaView wrapper if statusBarHeight is not given', () => {
    const tree = renderer
      .create(
        <Appbar.Header>
          <Appbar.Content title="Examples" />
        </Appbar.Header>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Uses a View wrapper if statusBarHeight is given a number', () => {
    const tree = renderer
      .create(
        <Appbar.Header statusBarHeight={100}>
          <Appbar.Content title="Examples" />
        </Appbar.Header>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
