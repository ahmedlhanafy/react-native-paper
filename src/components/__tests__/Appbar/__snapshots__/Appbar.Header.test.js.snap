// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppbarHeader Uses a SafeAreaView wrapper if statusBarHeight is not given 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "#6200ee",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
      "zIndex": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": 0,
        "paddingHorizontal": 4,
      }
    }
  >
    <View
      style={
        Object {
          "width": 48,
        }
      }
    />
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 12,
          },
          Array [
            false,
            Object {
              "alignItems": "center",
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        accessibilityRole="header"
        accessibilityTraits="header"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "fontSize": 17,
              },
              undefined,
            ],
          ]
        }
      >
        Examples
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 48,
        }
      }
    />
  </View>
</View>
`;

exports[`AppbarHeader Uses a View wrapper if statusBarHeight is given a number 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "zIndex": 0,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": 100,
        "paddingHorizontal": 4,
      }
    }
  >
    <View
      style={
        Object {
          "width": 48,
        }
      }
    />
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 12,
          },
          Array [
            false,
            Object {
              "alignItems": "center",
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        accessibilityRole="header"
        accessibilityTraits="header"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Helvetica Neue",
              },
              Object {
                "fontSize": 17,
              },
              undefined,
            ],
          ]
        }
      >
        Examples
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 48,
        }
      }
    />
  </View>
</View>
`;
